File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <algorithm>
#include <vector>

using namespace std;

int solution(vector<int> citations) {
  int answer = 0;

  sort(citations.begin(), citations.end());
  for (int i = 0;
       citations.end() - lower_bound(citations.begin(), citations.end(), i) >=
       i;
       i++) {
    answer = i;
  }
  return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.Arrays;

class Solution {

    public int solution(int[] citations) {
        Arrays.sort(citations);

        int left = 0, right = citations.length;
        while (left < right) {
            int mid = (left + right) / 2;
            if (citations[citations.length - mid - 1] >= mid + 1) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return left;
    }
}